### PR TITLE
Feature/33819 checkout step2 credit card

### PR DESCRIPTION
**Pull request by oseibonsu**

### CODE REVIEW

File: app/scripts/checkout/checkout.html [hunk 1]
```diff
@@ -39,8 +39,9 @@
 				<div class="th-checkout__step__title__stepName">Pay With</div>
 			</div>
 
-			<div ng-include=" 'pay-with/credit-card.html' "></div>
-			<div ng-include=" 'pay-with/credit-card-customer.html' "></div>
+			<div ng-include=" 'pay-with/credit-card-views.html' "></div>
+			<!--<div ng-include=" 'pay-with/credit-card.html' "></div>-->
```
Comment (prabinv) [inline app/scripts/checkout/checkout.html]: Can you remove the commented code and also the files if they are still part of the project?

Comment (oseibonsu) [inline app/scripts/checkout/checkout.html]: done

File: app/scripts/pay-with/credit-card-views-ctrl.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			// These are used to keep track of the card that is selected. selectedIndex is passed to the picker directive
+			_this.oldSelectedIndex = null;
+			_this.selectedIndex = null;
+
+			_this.init = function(){
+				_this.card = newCard();
+				if (_this.user.anonymous) {
+					_this.isUserCanEnterPaymentProfile = true;
+					_this.state = 'ENTERCARD';
+				}else{
```
Comment (prabinv) [inline app/scripts/pay-with/credit-card-views-ctrl.js]: Small nitpick - spaces. Pro-tip - In Webstorm, using the command `Ctrl+Alt+L` reformats the code for you.

Comment (oseibonsu) [inline app/scripts/pay-with/credit-card-views-ctrl.js]: done

File: app/scripts/pay-with/vn-credit-card-editor.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+						ret.push(i);
+					}
+					return ret;
+				};
+
+				scope.onExpMonthChanged = function (month, id) {
+					scope.card.expMonth = month.num;
+
+					// VN-PLACEHOLDER need a value and event ... set these for current button
+					var elem = angular.element(document.querySelector('#' + id));
```
Comment (prabinv) [inline app/scripts/pay-with/vn-credit-card-editor.js]: You don't need the `document.querySelector` here .... just doing `angular.element('#' + id)` would work just as fine.

File: app/scripts/pay-with/vn-credit-card-editor.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+					elem.focus();
+					elem.triggerHandler('change');
+
+				};
+
+				scope.onExpYearChanged = function (year, id) {
+					scope.card.expYear = year.toString().substring(2);
+
+					// VN-PLACEHOLDER need a value and event ... set these for current button
+					var elem = angular.element(document.querySelector('#' + id));
```
Comment (prabinv) [inline app/scripts/pay-with/vn-credit-card-editor.js]: You don't need the `document.querySelector` here .... just doing `angular.element('#' + id)` would work just as fine.

File: app/scripts/pay-with/vn-credit-card-picker.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+				scope.goBackFromEnter = function(){
+					scope.state = 'PICK';
+				};
+
+				scope.setIndex = function(id){
+					scope.selectedIndex = id;
+				};
+
+				scope.getCssClass = function(profile){
```
Comment (prabinv) [inline app/scripts/pay-with/vn-credit-card-picker.js]: This function is being used in multiple places. Might be a good candidate for a angular service / factory / value (key-value pairs)

**Pull request merged**